src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/CommonItemDialog.cs

```diff
@@ -0,0 +1,692 @@
+            // and to set defaults for some option flags.
+            // Initialize() is also called from the virtual
+            // Reset() function to restore defaults.
+            Initialize();
+        }
+
+        #endregion Constructors
+
+        //---------------------------------------------------
+        //
+        // Public Methods
+        //
+        //---------------------------------------------------
+        #region Public Methods
+
+        /// <summary>
+        ///  Resets all properties to their default values.
+        ///  Classes derived from FileDialog are expected to 
+        ///  call Base.Reset() at the beginning of their
+        ///  implementation of Reset() if they choose to
+        ///  override this function.
+        /// </summary>
+        public override void Reset()
+        {
+            Initialize();
+        }
+
+        #endregion Public Methods
+
+        //---------------------------------------------------
+        //
+        // Public Properties
+        //
+        //---------------------------------------------------
+        #region Public Properties
+
+        //   The actual flag is FOS_NODEREFERENCELINKS (set = do not dereference, unset = deref) - 
+        //   while we have true = dereference and false=do not dereference.  Because we expose
+        //   the opposite of the Windows flag as a property to be clearer, we need to negate 
+        //   the value in both the getter and the setter here.
+        /// <summary>
+        ///  Gets or sets a value indicating whether the dialog box returns the location 
+        ///  of the file referenced by the shortcut or whether it returns the location 
+        ///  of the shortcut (.lnk). Not all dialogs allow users to select shortcuts.
+        /// </summary>
+        public bool DereferenceLinks
+        {
+            get
+            {
+                return !GetOption(FOS.NODEREFERENCELINKS);
+            }
+            set
+            {
+
+                SetOption(FOS.NODEREFERENCELINKS, !value);
+            }
+        }
+
+        /// <summary>
+        ///  Gets or sets the initial directory displayed by the file dialog box.
+        /// </summary>
+        public string InitialDirectory
+        {
+            get
+            {
+                // Avoid returning a null string - return String.Empty instead.
+                return _initialDirectory.Value == null ? String.Empty : _initialDirectory.Value;
+            }
+            set
+            {
+
+                _initialDirectory.Value = value;
+            }
+        }
+
+        /// <summary>
+        ///  Restores the current directory to its original value if the user
+        ///  changed the directory while searching for files.
+        ///
+        ///  This property is only valid for SaveFileDialog;  it has no effect
+        ///  when set on an OpenFileDialog.
+        /// </summary>
+        public bool RestoreDirectory
```

Comment:
As per the new approved API we have to move this back to `FileDialog`. Can you move this back as well ?